File: src/languages/python.rs
pub mod python {
    use crate::language::language::{LanguageInformation, Language, execute_command};
    use crate::interpreted_language::interpreted_language::InterpretedLanguage;
    use crate::program_output::program_output::ProgramOutput;

    pub struct Python{
        info: LanguageInformation
    }

    impl Language for Python {
        fn from_config() -> Result<Self, String> where Self: Sized {
            let language_info_from_config: Result<LanguageInformation, String> = LanguageInformation::from_config("python3");

            return match language_info_from_config {
                Ok(info) => Ok(Python{ info }),
                Err(err) => Err(err)
            }
        }

        fn get_language_information(&self) -> &LanguageInformation {
            &self.info
        }

        fn run(&self, code_path: &str) -> Result<ProgramOutput, String> {
            execute_command(self.info.get_executable(), vec![code_path])
        }
    }
    impl InterpretedLanguage for Python {

    }
}

File: src/languages/python.py
"""
Module containing the Python language class.
"""
from subprocess import CompletedProcess

from src.config import Config
from src.interpreted_language import InterpretedLanguage
from src.language import execute_command
from src.language_information import LanguageInformation


class Python(InterpretedLanguage):
    """
    Class representing the Python language.
    """
    def __init__(self, config: Config):

        language_info = config.create_language_information('python3')

        self.__info: LanguageInformation = language_info

    def get_language_information(self) -> LanguageInformation:
        return self.__info

    def run(self, code_path: str) -> CompletedProcess:
        return execute_command(self.__info.executable, [code_path])
